x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/create_field/create_field.tsx

```diff
@@ -76,6 +104,25 @@ export const CreateField = React.memo(function CreateFieldComponent({
     }
   };
 
+  const [referenceFieldComboValue, setReferenceFieldComboValue] = useState<string>();
+  const [nameValue, setNameValue] = useState<string>();
+  const [inferenceIdComboValue, setInferenceIdComboValue] = useState<string>();
+  const [semanticFieldType, setSemanticTextFieldType] = useState<string>();
+
+  useFieldEffect(form, 'referenceField', setReferenceFieldComboValue);
+  useFieldEffect(form, 'inferenceId', setInferenceIdComboValue);
+  useFieldEffect(form, 'name', setNameValue);
+
+  const fieldTypeValue = form.getFields()?.type?.value as Array<{ value: string }>;
+  useEffect(() => {
+    if (fieldTypeValue === undefined || fieldTypeValue.length === 0) {
+      return;
+    }
+    setSemanticTextFieldType(
+      fieldTypeValue[0]?.value === 'semantic_text' ? fieldTypeValue[0].value : undefined
```

Comment:
I'm having a hard time understanding why we need to rely on the first element of the array? This seems a bit brittle 🤔

Reply:
There is only one item in fieldTypeValue. That's why we are selecting first item to get the type value.

```
fieldTypeValue = [{
    "value": "semantic_text",
    "label": "Semantic text"
}]
```